packages/aws-cdk/toolkit/lib/cli-io-host.ts

```diff
@@ -0,0 +1,118 @@
+import * as chalk from 'chalk';
+
+/**
+ * Basic message structure for toolkit notifications.
+ * Messages are emitted by the toolkit and handled by the IoHost.
+ *
+ * @param time The time the message was emitted.
+ * @param level The log level of the message.
+ * @param action The action that triggered the message.
+ * @param code A short code to identify the message.
+ * @param message The message text.
+ * @param forceStdout If true, the message will be written to stdout regardless of log level.
+ */
+interface IoMessage {
+  readonly time: Date;
+  readonly level: IoMessageLevel;
+  readonly action: IoAction;
+  readonly code: string;
+  readonly message: string;
+  readonly forceStdout?: boolean;
+}
+
+export type IoMessageLevel = 'error' | 'warn' | 'info' | 'debug' | 'trace';
+
+export type IoAction = 'synth' | 'list' | 'deploy' | 'destroy';
+
+/**
+ * Options for the CLI IO host.
+ *
+ * @param useTTY If true, the host will use TTY features like color.
+ * @param ci Flag representing whether the current process is running in a CI environment,
+ * If true, the host will write all messages to stdout.
+ */
+interface CliIoHostOptions {
+  useTTY?: boolean;
+  ci?: boolean;
```

Comment:
readonly here too. and again i am confused why the docstrings are not on the params themselves. this might very well work, but its not the code style i've come to expect in cdk.

Reply:
and defaults

Reply:
Defaults is a miss, I just decided to write it this way. Didn't know this was required, perhaps we should invest in a linting rule

Reply:
i suspect there to be a jsii reason for why we do things the other way. and there IS a linter rule that enforces that, its just not enabled on this package yet. Extra brownie points if you enable the linter on this package in a separate PR and start busting Momo for it too.